src/main/java/gg/projecteden/nexus/features/shops/ResourceWorldCommand.java

```diff
@@ -0,0 +1,491 @@
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static gg.projecteden.nexus.utils.ItemUtils.isNullOrAir;
+import static gg.projecteden.nexus.utils.WorldGroup.isResourceWorld;
+import static gg.projecteden.utils.StringUtils.prettyMoney;
+
+@NoArgsConstructor
+public class ResourceWorldCommand extends CustomCommand implements Listener {
+	private static final ResourceMarketLoggerService service = new ResourceMarketLoggerService();
+	private ResourceMarketLogger logger;
+
+	public ResourceWorldCommand(@NonNull CommandEvent event) {
+		super(event);
+		if (isPlayerCommandEvent())
+			logger = getLogger(world());
+	}
+
+	static {
+		World survival = Bukkit.getWorld("survival");
+		World resource = Bukkit.getWorld("resource");
+		if (survival != null && resource != null)
+			resource.setMonsterSpawnLimit((int) (survival.getMonsterSpawnLimit() * 1.5));
+	}
+
+	@Path
+	void warp() {
+		runCommand("warp resource");
+	}
+
+	@Confirm
+	@Permission("group.admin")
+	@Path("reset <test>")
+	void reset(boolean test) {
+		resetWorlds(test);
+	}
+
+	@Confirm
+	@Permission("group.admin")
+	@Path("setup <test>")
+	void setup(boolean test) {
+		setupWorlds(test);
+	}
+
+	private ResourceMarketLogger getLogger(World world) {
+		if (!isResourceWorld(world))
+			throw new InvalidInputException("Not allowed outside of resource world");
+
+		return service.get(world.getUID());
+	}
+
+	private void save(World world) {
+		service.queueSave(Time.SECOND.get(), getLogger(world));
+	}
+
+	@Async
+	@Confirm
+	@Path("logger add")
+	@Permission("group.admin")
+	void logger_add() {
+		WorldEditUtils worldEditUtils = new WorldEditUtils(player());
+		Region selection = worldEditUtils.getPlayerSelection(player());
+		if (selection.getArea() > 1000000)
+			error("Max selection size is 1000000");
+
+		for (Block block : worldEditUtils.getBlocks(selection))
+			logger.add(block.getLocation());
+
+		save(world());
+		send(PREFIX + "Added &e" + selection.getArea() + " &3locations to logger, new size: &e" + logger.size());
+	}
+
+	@Async
+	@Path("logger count")
+	@Permission("group.admin")
+	void logger_count() {
+		send(PREFIX + logger.size() + " coordinates logged");
+	}
+
+	@Async
+	@Environments(Env.TEST)
+	@Path("logger add random [amount]")
+	@Permission("group.admin")
+	void logger_add_random(@Arg("10000") int amount) {
+		if (isResourceWorld(world()))
+			throw new InvalidInputException("You must be in a resource world");
+
+		for (int i = 0; i < amount; i++) {
+			while (true) {
+				final Location location = getRandomLocation();
+				if (logger.contains(location))
+					continue;
+
+				logger.add(location);
+				break;
+			}
+		}
+
+		save(world());
+		send(PREFIX + "Added &e" + amount + " &3locations to logger, new size: &e" + getLogger(world()).size());
+	}
+
+	@NotNull
+	private Location getRandomLocation() {
+		final int x = RandomUtils.randomInt(-RADIUS, RADIUS);
+		final int y = RandomUtils.randomInt(-64, 319);
```

Comment:
```suggestion
		final int y = RandomUtils.randomInt(world().getMinHeight(), world().getMaxHeight()-1);
```

Reply:
I used these values to generate even greater randomness, to stress test the system more.